src/lib/api.js

```diff
@@ -1,6 +1,47 @@
 import axios from 'axios';
 
 const api = {
+    responseInterceptorId: null,
+    requestInterceptorId: null,
+    /**
+     * Adds a function to Axios that intercepts a response
+
+     * @public
+     * @param {Function} responseInterceptor - Function that gets called on each response
+     * @return {void}
+     */
+    addResponseInterceptor(responseInterceptor) {
+        if (responseInterceptor && typeof responseInterceptor === 'function') {
+            this.responseInterceptorId = axios.interceptors.response.use(responseInterceptor);
```

Comment:
It seems like this would add the interceptor to all future instances of Preview. Should we create a private instance of axios instead of operating on the global instance?

Reply:
If they are separate instances of `ContentPreview` they can have their own interceptors, if they are operating on the same instance with multiple `.show()` call they will have an issue. We can move to separate instance of Axios for each `show()` invocation, but it will be a larger refactor.

Reply:
Right. I'm not sure that refactor can be avoided as part of this feature enhancement, though. It sounds like we need to have `lib/api` return a class that creates its own private `axios` instance. Then each instance of Preview can have its own instance of `lib/api` and axios.